internal/crdao/services/event_tracking/track_voting_created.go

```diff
@@ -1,70 +1,335 @@
 package event_tracking
 
 import (
-	"casper-dao-middleware/internal/crdao/dao_event_parser/events"
+	"encoding/json"
+
 	"casper-dao-middleware/internal/crdao/di"
 	"casper-dao-middleware/internal/crdao/entities"
+	"casper-dao-middleware/internal/crdao/events"
 	"casper-dao-middleware/pkg/casper"
 	"casper-dao-middleware/pkg/casper/types"
+	"casper-dao-middleware/pkg/go-ces-parser"
 )
 
 type TrackVotingCreated struct {
 	di.EntityManagerAware
 
 	deployProcessed casper.DeployProcessed
-	eventBody       []byte
+	cesEvent        ces.Event
 }
 
 func NewTrackVotingCreated() *TrackVotingCreated {
 	return &TrackVotingCreated{}
 }
 
-func (s *TrackVotingCreated) SetEventBody(eventBody []byte) {
-	s.eventBody = eventBody
-}
-
 func (s *TrackVotingCreated) SetDeployProcessed(deployProcessed casper.DeployProcessed) {
 	s.deployProcessed = deployProcessed
 }
 
+func (s *TrackVotingCreated) SetCESEvent(event ces.Event) {
+	s.cesEvent = event
+}
+
 func (s *TrackVotingCreated) Execute() error {
-	votingCreated, err := events.ParseVotingCreatedEvent(s.eventBody)
+	var voting entities.Voting
+	var err error
+
+	switch s.cesEvent.Name {
+	case events.SimpleVotingCreatedEventName:
+		voting, err = s.newVotingFromSimpleVotingCreated()
+	case events.ReputationVotingCreatedEventName:
+		voting, err = s.newVotingFromReputationVotingCreated()
+	case events.RepoVotingCreated:
+		voting, err = s.newVotingFromRepoVotingCreated()
+	case events.SlashingVotingCreated:
+		voting, err = s.newVotingFromSlashingVotingCreated()
+	case events.KYCVotingCreated:
+		voting, err = s.newVotingFromKYCVotingCreated()
```

Comment:
Maybe let's split into separate tracking services?

It doesn't look like we reuse that much code here, but made the file too long.